src/segments/talosctl.go

```diff
@@ -0,0 +1,75 @@
+package segments
+
+import (
+	"errors"
+	"path/filepath"
+
+	"github.com/jandedobbeleer/oh-my-posh/src/platform"
+	"github.com/jandedobbeleer/oh-my-posh/src/properties"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	TALOSCTLNOACTIVECONFIG = "NO ACTIVE CONFIG FOUND"
+)
+
+type Talosctl struct {
+	props properties.Properties
+	env   platform.Environment
+
+	Context string
+}
+
+type TalosctlData struct {
+	ActiveContext string `yaml:"context"`
+}
+
+func (t *Talosctl) Template() string {
+	return " {{ .Context}} "
+}
+
+func (t *Talosctl) Init(props properties.Properties, env platform.Environment) {
+	t.props = props
+	t.env = env
+}
+
+func (t *Talosctl) Enabled() bool {
+	cfgDir := filepath.Join(t.env.Home(), ".talos")
+	configFile, err := t.getActiveConfig(cfgDir)
+	if err != nil {
+		t.env.Error(err)
+		return false
+	}
+
+	data, err := t.getTalosctlData(configFile)
+	if err != nil {
+		t.env.Error(err)
+		return false
+	}
+
+	if len(data.ActiveContext) == 0 {
```

Comment:
@JanDeDobbeleer I feel like this `if` check is somewhat superfluous, BUT it's more readable than:
```go
t.Context = data.ActiveContext
return len(data.ActiveContext) == 0
```
Since `t.Context` will be initialized as the empty string anyways, there is no harm in writing the above, but again, readability...